pybaseball/playerid_lookup.py

```diff
@@ -47,77 +50,116 @@ def chadwick_register(save: bool = False) -> pd.DataFrame:
 
 def get_lookup_table(save=False):
     table = chadwick_register(save)
-    #make these lowercase to avoid capitalization mistakes when searching
+    # make these lowercase to avoid capitalization mistakes when searching
     table['name_last'] = table['name_last'].str.lower()
     table['name_first'] = table['name_first'].str.lower()
     return table
 
 
-def playerid_lookup(last=None, first=None, player_list=None):
-    # force input strings to lowercase
-    if last:
-        last = last.lower()
-    if first:
-        first = first.lower()
-    table = get_lookup_table()
-
-    # if player_list has a value, then the user is passing in a list of players
-    # the list of players may be comma delimited for last, first, or just last
-    if player_list:
-        player_counter = 1
-        for player in player_list:
-            last = player.split(",")[0].strip()
-            first = None
-            if (len(player.split(",")) > 1):
-                first = player.split(",")[1].strip()
-            if (player_counter == 1):
-                results = playerid_lookup(last, first)
-            else:
-                results = results.append(playerid_lookup(last, first), ignore_index=True)
-            player_counter += 1
+def name_similarity(last: str, first: str, player_table: pd.DataFrame) -> pd.DataFrame:
+    """Calculates similarity of first and last name provided with all players in player_table
+
+    Args:
+        last (str): Provided last name
+        first (str): Provided first name
+        player_table (pd.DataFrame): Chadwick player table including names
+
+    Returns:
+        pd.DataFrame: 5 nearest matches from fuzzywuzzy.process
+    """
+    filled_df = player_table.fillna("")
+    chadwick_names = filled_df["name_first"] + " " + filled_df["name_last"]
+    fuzzy_matches = pd.DataFrame(process.extract(f"{first} {last}", chadwick_names, limit=5))
+    matched_names = fuzzy_matches[0].str.split(expand=True)
+    matched_names = matched_names.rename(columns={0: "name_first", 1: "name_last"})
+    return matched_names
+
+
+class player_search_client:
+    def __init__(self) -> None:
+        self.table = get_lookup_table()
+
+    def search(self, last: str = None, first: str = None, player_list: List[str] = None, fuzzy: bool = False) -> pd.DataFrame:
+        """Lookup playerIDs (MLB AM, bbref, retrosheet, FG) for a given player
```

Comment:
maybe we could use a pattern like the fangraphs queries, where we instantiate the client object and alias `player_search` to `player_search_client().search`?

https://github.com/jldbc/pybaseball/blob/master/pybaseball/datasources/fangraphs.py#L216

Reply:
IIRC the FG base class is low-cost and is executed every time. This one would be a module singleton. I think the model here is what we do with the Lahman ZipFile in memory.

Reply:
Simple outline:
```python3
_client = None
def player_search(...):
    global _client
    if not _client:
        _client = player_search_client()
    return _client.search(...)
```

Reply:
if I understand what you mean then I think the answer is no, the base class is instantiated only once, because of the way `import` keeps track of what's been imported. example, if you say 
```
from .datasources.fangraphs import fg_batting_data
```

then when you call `fg_batting_data()` it reuses the object it instantiated the first time

Reply:
I think what you suggested is fine, however, the only argument against it would be trying to avoid a `global`, which is mainly an argument about style

Reply:
For simplicity's sake, easier to do this with the global (especially since scope is limited)